backend/src/ast/converters/scratch/helpers.ts

```diff
@@ -0,0 +1,322 @@
+import {
+  ExpressionNode,
+  ExpressionNodeType,
+  LiteralNode,
+} from "src/ast/types/general-ast/ast-nodes/code-node/expression-node";
+import { Block } from "src/ast/types/input/scratch/generated/sb3";
+import {
+  BlockInputValue,
+  BlockTreeWithEventHatRoot,
+  NonHatBlock,
+  NonHatBlockTree,
+  TreeNode,
+} from "./types";
+import { isWhenBackdropSwitchesToBlock } from "src/ast/types/input/scratch/blocks/event/when-backdrop-switches-to";
+import { isWhenBroadcastReceivedBlock } from "src/ast/types/input/scratch/blocks/event/when-broadcast-received";
+import {
+  isWhenGreaterThanBlock,
+  WhenGreaterThanBlock,
+} from "src/ast/types/input/scratch/blocks/event/when-greater-than";
+import { isWhenKeyPressedBlock } from "src/ast/types/input/scratch/blocks/event/when-key-pressed";
+import { isWhenFlagClickedBlock } from "src/ast/types/input/scratch/blocks/event/when-flag-clicked";
+import { isWhenStageIsClickedBlock } from "src/ast/types/input/scratch/blocks/event/when-stage-is-clicked";
+import { iswhenThisSpriteIsClickedBlock } from "src/ast/types/input/scratch/blocks/event/when-this-sprite-is-clicked";
+import { isStartAsCloneBlock } from "src/ast/types/input/scratch/blocks/control";
+import { match, P } from "ts-pattern";
+import { BlockReferenceInput } from "src/ast/types/input/scratch/blocks/block-reference-input";
+import { BlockInputType } from "src/ast/types/input/scratch/block-input-type";
+import { isDefinitionBlock } from "src/ast/types/input/scratch/blocks/procedure/definition";
+import { convertInputsToExpression } from "./scratch-block-input-converter";
+import {
+  KnownBuiltinScratchCodeBlock,
+  KnownBuiltinScratchExpressionBlock,
+  KnownBuiltinScratchHatBlock,
+} from "src/ast/types/input/scratch/blocks";
+import {
+  isControlCodeBlock,
+  isControlExpressionBlock,
+} from "./scratch-control-block-converter";
+import {
+  isDataCodeBlock,
+  isDataExpressionBlock,
+} from "./scratch-data-block-converter";
+import {
+  isEventCodeBlock,
+  isEventExpressionBlock,
+} from "./scratch-event-block-converter";
+import {
+  isLooksCodeBlock,
+  isLooksExpressionBlock,
+} from "./scratch-looks-block-converter";
+import {
+  isMotionCodeBlock,
+  isMotionExpressionBlock,
+} from "./scratch-motion-block-converter";
+import { isOperatorExpressionBlock } from "./scratch-operator-block-converter";
+import {
+  isProcedureCodeBlock,
+  isProcedureExpressionBlock,
+} from "./scratch-procedure-block-converter";
+import {
+  isSensingCodeBlock,
+  isSensingExpressionBlock,
+} from "./scratch-sensing-block-converter";
+import {
+  isSoundCodeBlock,
+  isSoundExpressionBlock,
+} from "./scratch-sound-block-converter";
+
+export const createLiteralNode = (type: string, value: string): LiteralNode => {
+  return {
+    nodeType: AstNodeType.code,
+    codeType: CodeNodeType.expression,
+    expressionType: ExpressionNodeType.literal,
+    type,
+    value,
+  };
+};
+
+const createArgumentsFromInputsAndFields = <TKey extends string>(
```

Comment:
This function is hard to read. Anything we could do to improve it / document it?